universal-login-sdk/lib/api/DeployedWallet.ts

```diff
@@ -99,8 +104,7 @@ export class DeployedWallet implements ApplicationWallet {
     }
 
     const execution = await this.sdk.addKeys(this.contractAddress, addresses, this.privateKey, {gasToken: ETHER_NATIVE_TOKEN.address, gasPrice: DEFAULT_GAS_PRICE, gasLimit: DEFAULT_GAS_LIMIT});
-    await execution.waitToBeSuccess();
-    return codes;
+    return {codes, execution};
```

Comment:
How about:
```
return {
    waitToBeSuccess: async () => {
        await execution.waitToBeSuccess();
        return codes;
  }
};
```

Reply:
👍

Reply:
That would mean that we'd need to always wait for execution to succeed to get the backup codes

Reply:
And that is correct because we don't want to have codes that are not in the contract.